File: Basic Programming/Hackerrank 30 days of code c++ solution/Day_29.cpp
/*                    Name : Arif Khan
                      Judge: HACKERRANK
                      University: Primeasia University
                      problem: Day 29: Bitwise AND
                      Difficulty: Easy
                      Problem Link: https://www.hackerrank.com/challenges/30-bitwise-and/problem
*/

#include <bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin >> t;
    for(int a0 = 0; a0 < t; a0++){
        int n, k,ans = 0,m = 0;
        cin >> n >> k;
        for(int i = 1; i<= n ;i++){
              for(int j = i+1; j<=n ;j++){
                ans = i & j;
                if(k > ans and m < ans)
                    m = ans;
            }
        }
        cout<<m<<endl;
    }
    return 0;
}

File: Basic Programming/Hackerrank 30 days of code python3 solution/Day_29.py
#                      Name : Arif Khan
#                      Judge: HACKERRANK
#                      University: Primeasia University
#                      problem: Day 29: Bitwise AND
#                      Difficulty: Easy
#                      Problem Link: https://www.hackerrank.com/challenges/30-bitwise-and/problem
#

import sys



if __name__ == '__main__':
    t = int(input())

    for t_itr in range(t):
        nk = input().split()

        n = int(nk[0])

        k = int(nk[1])

        mx = 0
        f = False
        for i in reversed(range(k)):
            for j in range(10):
                m = 2**j
                if m != (i&m) and (i+m) <= n:
                    mx = i
                    f = True
                    break
            if f:
                break
        
        print(mx)